twoliter/embedded/imghelper

```diff
@@ -0,0 +1,398 @@
+
+    # Replace the embedded vendor certificate, then sign shim with the db key.
+    pesign -i "${shim}" -l
+    objdumpcopy "${shim}" \
+      --update-section ".vendor_cert=${SBKEYS}/vendor.obj"
+    pesign -i "${shim}" -o "${shim}.signed" -s "${shim_sign_key[@]}"
+    mv "${shim}.signed" "${shim}"
+    pesign -i "${shim}" -l
+    pesigcheck -i "${shim}" -n 0 -c "${SBKEYS}/db.cer"
+
+    # Remove the original key if necessary, then sign the MOK manager as well.
+    if [[ "$(pesign -i "${mokm}" -l)" == "No signatures found." ]]; then
+      mv "${mokm}" "${mokm}.unsigned"
+    else
+      pesign -i "${mokm}" -o "${mokm}.unsigned" -u 0 -r
+    fi
+    pesign -i "${mokm}.unsigned" -o "${mokm}.signed" -s "${code_sign_key[@]}"
+    mv "${mokm}.signed" "${mokm}"
+    rm "${mokm}.unsigned"
+    pesign -i "${mokm}" -l
+    pesigcheck -i "${mokm}" -n 0 -c "${SBKEYS}/vendor.cer"
+
+    # Remove the original vendor key if necessary, replace the embedded gpg
+    # public key, then sign grub with the vendor key.
+    if [[ "$(pesign -i "${grub}" -l)" == "No signatures found." ]]; then
+      mv "${grub}" "${grub}.unsigned"
+    else
+      pesign -i "${grub}" -o "${grub}.unsigned" -u 0 -r
+    fi
+    objdumpcopy "${grub}.unsigned" \
+      --file-alignment 4096 \
+      --update-section ".pubkey=${SBKEYS}/config-sign.pubkey"
+    pesign -i "${grub}.unsigned" -o "${grub}.signed" -s "${code_sign_key[@]}"
+    mv "${grub}.signed" "${grub}"
+    rm "${grub}.unsigned"
+    pesign -i "${grub}" -l
+    pesigcheck -i "${grub}" -n 0 -c "${SBKEYS}/vendor.cer"
+  else
+    # Generate a zero-sized certificate in the expected format.
+    cert_table /dev/null "${SBKEYS}/vendor.obj"
+
+    # Replace the embedded vendor certificate with the zero-sized one, which
+    # shim will ignore when Secure Boot is disabled.
+    objdumpcopy "${shim}" \
+      --update-section ".vendor_cert=${SBKEYS}/vendor.obj"
+
+    # Remove the embedded gpg public key to disable GRUB's signature checks.
+    objdumpcopy "${grub}" \
+      --file-alignment 4096 \
+      --remove-section ".pubkey"
+  fi
+}
+
+vmlinuzsetup() {
+  local vmlinuz
+  vmlinuz="${1:?}"
+  if [[ "$(pesign -i "${vmlinuz}" -l)" == "No signatures found." ]]; then
+    mv "${vmlinuz}" "${vmlinuz}.unsigned"
+  else
+    pesign -i "${vmlinuz}" -o "${vmlinuz}.unsigned" -u 0 -r
+  fi
+  pesign -i "${vmlinuz}.unsigned" -o "${vmlinuz}.signed" -s "${code_sign_key[@]}"
+  mv "${vmlinuz}.signed" "${vmlinuz}"
+  rm "${vmlinuz}.unsigned"
+  pesign -i "${vmlinuz}" -l
+  pesigcheck -i "${vmlinuz}" -n 0 -c "${SBKEYS}/vendor.cer"
+}
+
+vmlinuzhmac(){
+  local vmlinuz
+  vmlinuz="${1:?}"
+  openssl sha512 -hmac FIPS-FTW-RHT2009 -hex "${vmlinuz}" \
+  | awk -v vmlinuz="${vmlinuz}" '{ print $2 "  " vmlinuz }' \
+  > ".${vmlinuz}.hmac"
+}
+
+sign_grub(){
+  local grub_cfg
+  grub_cfg="${1:?}"
+  gpg --batch --no-tty --detach-sign "${grub_cfg}"
+  gpg --batch --no-tty --verify "${grub_cfg}.sig" "${grub_cfg}"
+}
```

Comment:
These functions implicitly assume that the current working directory is the same as the `vmlinuz` or `grub.cfg` that they're operating. It would be good to correct that - take a full path to those artifacts and then use `"${vmlinuz##*/}"` to compute the target directory.